From 406b2f92b586dcfa1e9aa23af3277c31dc0a980d Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Tue, 25 Feb 2014 16:56:49 +0100
Subject: [PATCH 8/9] suspend-on-idle: Ensure we still time out if a stream
 remains corked

If a stream is started corked and remains corked, the sink/source
remained idle without being properly suspended. This patch fixes
that issue.

BugLink: https://bugs.launchpad.net/bugs/1284415
Tested-by: Ricardo Salveti <ricardo.salveti@canonical.com>
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/modules/module-suspend-on-idle.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/module-suspend-on-idle.c b/src/modules/module-suspend-on-idle.c
index a284f858..a02712ba 100644
--- a/src/modules/module-suspend-on-idle.c
+++ b/src/modules/module-suspend-on-idle.c
@@ -125,7 +125,7 @@ static pa_hook_result_t sink_input_fixate_hook_cb(pa_core *c, pa_sink_input_new_
 
     if ((d = pa_hashmap_get(u->device_infos, data->sink))) {
         resume(d);
-        if (pa_sink_check_suspend(d->sink, NULL, NULL) <= 0)
+        if (d->sink && pa_sink_check_suspend(d->sink, NULL, NULL) <= 0)
             restart(d);
     }
 
-- 
2.11.1

